docs/datatypes/stm.md

```diff
@@ -0,0 +1,115 @@
+---
+id: datatypes_stm
+title:  "STM"
+---
+
+Software Transactional Memory (STM) is a technique which allows composition of arbitrary atomic operations. It is the 
+software analog of transactions in database systems. STM can perform groups of memory operations in an atomic fashion. 
+The API is based on `Control.Concurrent.STM` although the implementation is completely different.
+
+STM brings several advantages over traditional low-level locking mechanisms:
+* **Compositionality**
+* Freedom from deadlocks
+* Automatic rollback on exceptions or timeouts
+* Freedom from tension between lock granularity and concurrency
+
+The main problem with traditional concurrent programs using locks is that they do not compose and correct fragments
+can fail when they are combined as the order in which locks are acquired and released can easily cause deadlocks as 
+more locks are introduced into the equation.
+
+Transactional memory eliminates many of the difficulties that affect low-level lock-based programming since transactional
+data structures are lock-free.
+
+## Transactional data structures
+
+There are a variety of transactional data structures that can take part in an STM transaction. These include:
```

Comment:
The second part of the sentence feels redundant.

Reply:
woops added